chia-bls/src/cached_bls.rs

```diff
@@ -0,0 +1,275 @@
+extern crate lru;
+use lru::LruCache;
+use std::num::NonZeroUsize;
+use std::borrow::Borrow;
+use crate::Signature;
+use crate::hash_to_g2;
+use crate::aggregate_verify as agg_ver;
+use crate::gtelement::GTElement;
+use crate::PublicKey;
+use std::collections::HashMap;
+use sha2::{Digest, Sha256};
+
+#[cfg(feature = "py-bindings")]
+use pyo3::{pyclass, pymethods, PyResult};
+#[cfg(feature = "py-bindings")]
+use pyo3::types::{PyList, PyBool};
+
+pub type Bytes32 = [u8; 32];
+pub type Bytes48 = [u8; 48];
+
+#[cfg_attr(
+    feature = "py-bindings",
+    pyclass(name = "BLSCache"),
+)]
+pub struct BLSCache {
+    cache: LruCache<Bytes32, GTElement>,
+}
+
+impl BLSCache {
+    
+    pub fn generator(cache_size: Option<usize>) -> Self {
+        let cache: LruCache<Bytes32, GTElement> = LruCache::new(NonZeroUsize::new(cache_size.unwrap_or(50000)).unwrap());
```

Comment:
I wasn't familiar with `NonZeroUsize`. Seems a bit strange they'd use that in their API. Maybe too clever by half.

Note that using `.unwrap` is a bad smell because it panics on failure. There's probably a way to do this without `.unwrap`. Maybe just do `cache_size = cache_size | 50000`.

```
    pub fn generator(cache_size: usize) -> Self {
        let cache: LruCache<Bytes32, GTElement> = LruCache::new(NonZeroUsize::new(cache_size | 50000).unwrap());
        Self{cache}
    }
```

I haven't test this.